source/SavedGame.cpp

```diff
@@ -57,7 +61,7 @@ void SavedGame::Load(const string &path)
 					break;
 				}
 		}
-		else if(node.Token(0) == "ship" && !shipSprite)
+		else if(node.Token(0) == "ship" && !shipSprite && ++flagshipIterator == flagshipTarget)
```

Comment:
If anything, that's more accurate. If a player somehow has a ship with no sprite as their flagship, they are certainly breaking the game, but it doesn't stop being their flagship.

Reply:
What if there is no "flagship" index? Then the first ship is used but it might not have a sprite => whereas on master it would use the second ship.

But anyways, then you should remove the code related to the sprite thingy if it isn't used anymore. Cause right now it's dead code.